### PR TITLE
Upgrade puppeteer to 5.5.0

### LINKED ISSUE #23333: Scripts: Upgrade Puppeteer to next major version (5.x)
Release notes: ~https://github.com/puppeteer/puppeteer/releases/tag/v4.0.0~ https://github.com/puppeteer/puppeteer/releases/tag/v5.3.1

In case of `@wordpress/scripts`, it will require updating the existing alias to `puppeteer-core`. The same alias is also listed in the root `package.json` file.

There are also related packages that might need to be updated in other places like `jest-puppeteer`.

### PR DESCRIPTION
**Pull request by kevin940726**
## Description

Close #23333.

## How has this been tested?

`npm run test-e2e`

## Types of changes

Breaking change

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/e2e-test-utils/src/change-site-timezone.js [hunk 1]
```diff
@@ -21,7 +21,10 @@ export async function changeSiteTimezone( timezone ) {
 		( element ) => element.options[ element.selectedIndex ].text
 	);
 	await page.select( '#timezone_string', timezone );
-	await page.click( '#submit' );
+	await Promise.all( [
+		page.click( '#submit' ),
+		page.waitForNavigation( { waitUntil: 'networkidle0' } ),
+	] );
```
Comment (kevin940726) [inline packages/e2e-test-utils/src/change-site-timezone.js:27]: This is to fix an error similar to #27358 becoming more consistent after upgrading puppeteer.

The solution is to use [`waitForNavigation`](https://github.com/puppeteer/puppeteer/blob/v5.5.0/docs/api.md#pagewaitfornavigationoptions) for places that will trigger a page navigation.

**Review by gziolo — APPROVED** [review]
Cool, thank you for working on it. I'm surprised that only one test needed an update.

Can you include a note about the breaking change since it's a new major version of `puppeteer`. It can be similar to what we had in the past:
https://github.com/WordPress/gutenberg/blob/master/packages/scripts/CHANGELOG.md#breaking-changes-3

**Comment by kevin940726** [comment]
@gziolo Sure! Just added in 1f35c2c.

**Comment by gziolo** [comment]
I merged this PR. The only failing e2e test is a known issue. Great work bringing Puppeteer to the latest version ❤️

**Pull request merged**